internal/gatewayapi/route.go

```diff
@@ -1848,11 +1846,15 @@ func (t *Translator) processBackendDestinationSetting(name string, backendRef gw
 	return ds
 }
 
-func serviceAppProtocolToIRAppProtocol(ap string, defaultProtocol ir.AppProtocol) ir.AppProtocol {
-	switch ap {
-	case "kubernetes.io/h2c":
+// serviceAppProtocolToIRAppProtocol translates the appProtocol string into an ir.AppProtocol.
+//
+// When supportLegacy is enabled, `grpc` will be parsed as a valid option for HTTP2.
+// See https://github.com/envoyproxy/gateway/issues/5485#issuecomment-2731322578.
+func serviceAppProtocolToIRAppProtocol(ap string, defaultProtocol ir.AppProtocol, supportLegacy bool) ir.AppProtocol {
```

Comment:
this's not legacy, compatibility sounds better?

Reply:
True, how does `grpcCompatibility` sound?